src/openforms/js/components/admin/form_design/registrations/objectsapi/fields/CatalogueSelect.js

```diff
@@ -0,0 +1,90 @@
+import {useFormikContext} from 'formik';
+import {FormattedMessage} from 'react-intl';
+import useAsync from 'react-use/esm/useAsync';
+
+import Field from 'components/admin/forms/Field';
+import FormRow from 'components/admin/forms/FormRow';
+import ReactSelect from 'components/admin/forms/ReactSelect';
+import {get} from 'utils/fetch';
+
+const ENDPOINT = '/api/v2/registration/plugins/objects-api/catalogues';
+
+const getCatalogues = async apiGroupID => {
+  const response = await get(ENDPOINT, {objects_api_group: apiGroupID});
+  if (!response.ok) {
+    throw new Error('Loading available object type versions failed');
+  }
+  const catalogues = response.data;
+
+  const _optionsByRSIN = {};
+  for (const catalogue of catalogues) {
+    const {rsin} = catalogue;
+    if (!_optionsByRSIN[rsin]) _optionsByRSIN[rsin] = [];
+    _optionsByRSIN[rsin].push(catalogue);
+  }
+
+  const groups = Object.entries(_optionsByRSIN)
+    .map(([rsin, options]) => ({
+      label: rsin,
+      options: options.sort((a, b) => a.label.localeCompare(b.label)),
+    }))
+    .sort((a, b) => a.label.localeCompare(b.label));
+
+  return groups;
+};
+
+const CatalogueSelect = () => {
+  const {
+    values: {objectsApiGroup = null, catalogue = {}},
+    getFieldHelpers,
+  } = useFormikContext();
+
+  const {setValue} = getFieldHelpers('catalogue');
+
+  const {
+    loading,
+    value: optionGroups = [],
+    error,
+  } = useAsync(async () => {
+    if (!objectsApiGroup) return [];
+    return await getCatalogues(objectsApiGroup);
+  }, [objectsApiGroup]);
+  if (error) throw error;
```

Comment:
I noticed that the entire form crashes if the catalogi API can't be reached

![Screenshot from 2024-08-08 09-23-33](https://github.com/user-attachments/assets/1cdc1fe3-93bd-407c-b319-7be43a4c194b)

Reply:
oh that's bad, I will add some more error boundaries!